node/src/actors/rad_manager/mod.rs

```diff
@@ -8,12 +8,45 @@
 
 use crate::utils::stop_system_if_panicking;
 
+use core::iter;
+use itertools::Itertools;
+
 mod actor;
 mod handlers;
 
 /// RadManager actor
 #[derive(Default)]
-pub struct RadManager;
+pub struct RadManager {
+    /// Addresses of proxies to be used as extra transports when performing data retrieval.
+    proxies: Vec<String>,
+}
+
+impl RadManager {
+    /// Register a new proxy address to be used for "paranoid retrieval", i.e. retrieving data
+    /// sources through different transports so as to ensure that the data sources are consistent
+    /// and we are taking as small of a risk as possible when committing to specially crafted data
+    /// requests that may be potentially ill-intended.
+    pub fn add_proxy(&mut self, proxy_address: String) {
+        self.proxies.push(proxy_address)
+    }
+
+    /// Derive HTTP transports from registered proxies.
+    ///
+    /// This internally injects a `None` at the beginning, standing for the base "clearnet"
+    /// transport (no proxy).
```

Comment:
I have now added such possibility through the `connections.unproxied_retrieval` setting, which defaults to `true` standing for enabled unproxied retrievals. `false` will disable them. In that case, proxy addresses need to be provided, otherwise it will  panic at start. This logic is covered by tests.